site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2437,6 +2439,53 @@ public function getGradedComponentsCountByGradingSections($g_id, $sections, $sec
         return $return;
     }
 
+
+    /**
+     * Returns an array of Verified components for each Section
+     *
+     * @param  array<int>  $sections
+     * @return array<int|string,int>
+     */
+    public function getVerifiedComponentsCountByGradingSections(string $g_id, array $sections, string $section_key, bool $is_team): array {
+        $unit = "users";
+        $id = "user_id";
+        if ($is_team) {
+            $unit = "gradeable_teams";
+            $id = "team_id";
+        }
+
+        $this->course_db->query(
+            "
+            SELECT 
+                " . $unit . "." . $section_key . ", COUNT(" . $unit . "." . $id . ") as verified_components_count
+            FROM 
+                gradeable_data as gd
+                JOIN gradeable_component_data as gcd ON (gd.gd_id = gcd.gd_id)
+                JOIN " . $unit . " ON (gd.gd_" . $id . " = " . $unit . "." . $id . ")
+            WHERE
+                gd.g_id = ?
+                AND CAST (" . $unit . "." . $section_key . " AS TEXT) = ANY (ARRAY[" . implode(",", array_map(
+                    function ($v) {
+                        return "'" . $v . "'";
+                    },
+                    $sections
```

Comment:
`$sections` is user-controlled, and should not be inserted directly into a query.  Use a prepared statement instead.  `$this->createParameterList)` will also be useful to generate a list of question marks.